### ISSUE TITLE
A sniff to identify missing i18n within HTML

### ISSUE THREAD

**Issue by johnbillion** [WordPress/WordPress-Coding-Standards#1258]
This feature request might be very difficult to implement, or it might be partially possible, which could be better than nothing.

Given the following code, I'd like a sniff which identifies the fact that text is output which does not pass through an i18n function and is therefore not localisable.

```php
function my_function() {
	?>
	<strong>Hello</strong>
	<?php
}

my_function();
```

Does this sound possible?

**Comment by JDGrimes** [comment]
I think it would be possible to do this with a sniff, though a few years ago a wrote a stand-alone tool that does something similar: https://github.com/JDGrimes/wp-l10n-validator

One issue with doing this in a sniff is the difficulty of filtering through potential false-positives. In that tool I built, I  just make it easy to ignore all remaining issues after you've fixed those that are actually a problem. Then it keeps that cache of the ignores for you and won't tell you about them again, as you continue to scan your code (like with CI). It isn't perfect, but it was easier than coming up with rules to reduce false-positives, as some are hard to detect.
